From 2024439bd5ceb145eeeb428b2a59e9b905153ac3 Mon Sep 17 00:00:00 2001
Message-Id: <2024439bd5ceb145eeeb428b2a59e9b905153ac3.1663577849.git.geliang.tang@suse.com>
From: Florian Westphal <fw@strlen.de>
Date: Thu, 11 Aug 2022 13:30:39 +0200
Subject: [PATCH] netfilter: nf_tables: fix scheduling-while-atomic splat
Git-commit: 2024439bd5ceb145eeeb428b2a59e9b905153ac3
Patch-mainline: v6.0-rc2
References: bsn#338
Modified-by-SEL: No

nf_tables_check_loops() can be called from rhashtable list
walk so cond_resched() cannot be used here.

Fixes: 81ea01066741 ("netfilter: nf_tables: add rescheduling points during loop detection walks")
Signed-off-by: Florian Westphal <fw@strlen.de>
Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 net/netfilter/nf_tables_api.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/net/netfilter/nf_tables_api.c b/net/netfilter/nf_tables_api.c
index 989c9782ecc3..d90434eadc1b 100644
--- a/net/netfilter/nf_tables_api.c
+++ b/net/netfilter/nf_tables_api.c
@@ -9426,13 +9426,9 @@ static int nf_tables_check_loops(const struct nft_ctx *ctx,
 				break;
 			}
 		}
-
-		cond_resched();
 	}
 
 	list_for_each_entry(set, &ctx->table->sets, list) {
-		cond_resched();
-
 		if (!nft_is_active_next(ctx->net, set))
 			continue;
 		if (!(set->flags & NFT_SET_MAP) ||
-- 
2.35.3

